### PR TITLE
Worker refactor

### PR DESCRIPTION
**Pull request by mwang87**
Fixes: https://github.com/mwang87/MetabolomicsSpectrumResolver/issues/141

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -8,11 +8,44 @@ services:
...
+      context: .
+      dockerfile: Dockerfile
+    container_name: metabolomicsusi-worker
+    volumes:
+      - ./temp:/app/temp:rw
+      - ./logs:/app/logs:rw
+    command: /app/run_worker.sh
+    restart: unless-stopped
+    depends_on: 
+      - metabolomicsusi-redis
```
Comment (bittremieux) [inline docker-compose.yml:29]: Do we need both RabbitMQ and Redis? Don't they do the same thing?

**Review by bittremieux — APPROVED** [review]
I made a few minor formatting changes, but it looks good. Just two small remarks.

I usually just run the app directly rather than in Docker, so I didn't run it yet with this code because I haven't set up a Redis service yet.

File: docker-compose.yml (hunk 1 above)
Comment (mwang87) [reply docker-compose.yml:29]: So broadly the same thing but the scale at which its done is different. rabbitmq presents problems when the number of results gets large, but redis should be fine I think. The only advantage of rabbitmq is on restart to preserve the queue

File: Makefile [hunk 2]
```diff
@@ -16,7 +16,8 @@ interactive: clean
 bash: clean
 	docker run -it -p 5087:5000 --name metabolomicsusi metabolomicsusi bash
 
-
+clear-cache:
+	sudo rm temp/joblibcache -rf
```
[the remark this answers is not in the document]
Comment (mwang87) [reply Makefile:20]: Yes, I agree lets use tmp.

**Comment by mwang87** [comment]
TODO:

1. Move to redis only
2. Move temp to tmp

**Comment by bittremieux** [comment]
Made those small changes; it runs nice and fast. 🙂

Right now we fall back to direct resolving in `views.py` if the Celery asynchronous tasks don't work, but it takes a long time to hit that timeout. Maybe we should just remove direct resolving as a fallback and report an error instead? Because in that case something is wrong which needs to be fixed. Waiting so long is not a good user experience, but users might be more forgiving if they know something is wrong.
Also, I don't think we really log it right now if Celery is not working and we use the fallback option, so how do we know that something is wrong?

We can potentially still use `/tmp` rather than a local `tmp` directory to keep the working directory cleaner. It doesn't really make that much difference.

This will be super useful though to process requests faster.

**Pull request merged**